raqm.c

```diff
@@ -457,30 +461,34 @@ raqm_shape_u32 (unsigned int* u32_str,
                 raqm_direction_t direction)
 {
     int i = 0;
-    FriBidiCharType* types;
-    FriBidiLevel* levels;
-    FriBidiParType par_type;
-    hb_script_t* scripts;
-    hb_unicode_funcs_t* unicode_funcs;
-    hb_script_t last_script_value;
-    int last_script_index = -1;
-    int last_set_index = -1;
-    Stack* script_stack;
+    int j = 0;
```

Comment:
This variable is only used in the for loop below, so it is better be declared there. You just need to move it to the top of the block to fix the compiler warning.